Core/src/IfcGeometryConverter/ProfileConverter.h

```diff
@@ -469,8 +469,63 @@ namespace OpenInfraPlatform {
 			void convertIfcOpenCrossProfileDef(const EXPRESSReference<typename IfcEntityTypesT::IfcOpenCrossProfileDef>& profileDef,
 				std::vector<std::vector<carve::geom::vector<2>>>& paths) const
 			{
-				throw oip::UnhandledException(profileDef);
+				if (profileDef->expired())
+					throw oip::ReferenceExpiredException(profileDef);
+
+				std::vector<carve::geom::vector<2>> outer_loop;
+
+				bool horizontal = profileDef->HorizontalWidths;
+
+				std::vector<double> widths;
+				widths.reserve(profileDef->Widths.size());
+				std::transform(
+					profileDef->Widths.begin(),
+					profileDef->Widths.end(),
+					widths.begin(),
+					[](auto& it) {return it * UnitConvert()->getLengthInMeterFactor(); });
+
+				std::vector<double> slopes;
+				slopes.reserve(profileDef->Slopes.size());
+				std::transform(
+					profileDef->Slopes.begin(),
+					profileDef->Slopes.end(),
+					slopes.begin(),
+					[](auto& it) {return it * UnitConvert()->getAngleInRadianFactor(); });
+
+				if (widths.size() != slopes.size()) {
+					throw oip::InconsistentModellingException(profileDef, "Number of Widths is not equal to the number of Slopes");
+				}
+				double TagX = 0.0;
+				double TAgY = 0.0;
+				double x, y;
+
+				for (int i = 0; i < widths.size(); i++) {
+					paths.push_back(carve::geom::VECTOR(TagX, TagY));
+					std::tie(x, y) = CalculateXYFromPolar(horizontal, widths[i], slopes[i]);

```

Comment:
Following @hechth suggestion:

Instead of `widths[i]` one can use `profileDef->Widths[i] * UnitConvert()->...`, similarly of `slopes[i]`. In this way, there is no need for `std::vector`s and the transformations. Please change accordingly.

Reply:
Resolved: 5c1d3b6